analyses/cell-type-neuroblastoma-04/exploratory-notebooks/scanvi-singler-comparison.Rmd

```diff
@@ -0,0 +1,391 @@
+We'll look at the median value for each as well as the number of cells that were annotated for each label:
+
+```{r}
+scanvi_df |>
+  dplyr::group_by(scanvi) |>
+  dplyr::summarize(
+    median_pp = median(scanvi_posterior), 
+    total_cells = dplyr::n()
+  ) |>
+  dplyr::arrange(median_pp)
+```
+
+There are very few cells in general associated with the cell types that have very low median posterior probabilities (`NKT cell` and `Migratory cDC`), which is comforting.
+
+A `0.9` threshold for the posterior probability here seems reasonable given how high these values are in general. 
+We'll therefore recode cells with posterior probabilities below 0.9 to `unknown_scanvi`.
+
+```{r}
+pp_threshold <- 0.9
+scanvi_df <- scanvi_df |>
+  dplyr::mutate(scanvi = ifelse(
+    scanvi_posterior < pp_threshold, 
+    "unknown_scanvi",
+    scanvi
+  ))
+```
+
+## Compare labels
+
+We'll create a single data frame with UMAP coordinates and all cell types (consensus, `SingleR`, and `scANVI`):
+
+```{r}
+celltype_df <- scuttle::makePerCellDF(
+  merged_sce,
+  use.coldata = c("sample_id", "consensus_celltype_annotation"),
+  use.dimred = c("UMAP")
+) |>
+  # there ARE repeated barcodes so we need to keep cell_id around
+  tibble::rownames_to_column("cell_id") |>
+  dplyr::rename(
+    UMAP1 = UMAP.1,
+    UMAP2 = UMAP.2,
+    consensus_annotation = consensus_celltype_annotation
+  ) |>
+  dplyr::left_join(validation_df, by = "consensus_annotation") |>
+  # recode consensus NAs to "unknown_consensus"
+  dplyr::mutate(
+    validation_group_annotation = ifelse(
+      is.na(validation_group_annotation), "unknown_consensus", validation_group_annotation
+    )
+  ) |>
+  dplyr::select(-consensus_annotation) |>
+  # merge in the annotations
+  dplyr::left_join(singler_df, by = "cell_id") |>
+  dplyr::left_join(scanvi_df, by = "cell_id") |>
+  # make it tidy
+  tidyr::pivot_longer(
+    c(validation_group_annotation, singler, scanvi), 
+    names_to = "celltype_method", 
+    values_to = "label"
+  ) |>
+  harmonize_celltypes(label, label_recoded) 
+```
+
+
+## UMAP
+
+This section displays the UMAP for the _merged object_ (not integrated) with cell type labels harmonized for comparison.
+We show a panel for each cell type method.
+
+```{r, fig.width = 12, fig.height = 8}
+ggplot(celltype_df) +
+  aes(x = UMAP1, y = UMAP2, color = label_recoded) +
+  geom_point(alpha = 0.5, size = 0.2) +
+  scale_color_manual(values = recoded_celltype_pal) +
+  facet_wrap(vars(celltype_method)) +
+  coord_equal() +
+  theme(
+    legend.position = "bottom", 
+    axis.text = element_blank(), 
+    axis.ticks = element_blank()
+  ) +
+  guides(color = guide_legend(override.aes = list(size = 2, alpha = 1)))
+```
+
+
+## Heatmap
+
+This section shows a stacked heatmap comparing the `SingleR` and `scANVI` annotations to the consensus validation groups.
```

Comment:
Didn't we annotate these with the immune zoom version of the atlas? I thought in other plots you've shown you get different types of NK cells and different types of T cells but I don't see that here? I think if this plot is summarizing the comparison to consensus that's fine to keep in, but we also want a plot that looks at all of the originally assigned `NBAtlas` labels comparing `scANVI` to `SingleR`.